lib/RoadizCoreBundle/src/Bag/DecoratedNodeTypes.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+namespace RZ\Roadiz\CoreBundle\Bag;
+
+use RZ\Roadiz\Bag\LazyParameterBag;
+use RZ\Roadiz\Contracts\NodeType\NodeTypeResolverInterface;
+use RZ\Roadiz\CoreBundle\Entity\NodeType;
+use RZ\Roadiz\CoreBundle\Repository\NodeTypeDecoratorRepository;
+
+/**
+ * @method NodeType|null get(string $key, $default = null)
+ */
+final class DecoratedNodeTypes extends LazyParameterBag implements NodeTypeResolverInterface
+{
+    public function __construct(
+        private readonly NodeTypes $nodeTypesBag,
+        private readonly NodeTypeDecoratorRepository $repository,
+    ) {
+        parent::__construct();
+    }
+
+    protected function populateParameters(): void
+    {
+        $nodeTypes = $this->nodeTypesBag->all();
+        foreach ($nodeTypes as $nodeType) {
+            $decoratedNodeType = clone $nodeType;
+            $nodeTypeDecorators = $this->repository->findByNodeType($decoratedNodeType);
+            foreach ($nodeTypeDecorators as $nodeTypeDecorator) {
+                $nodeTypeDecorator->applyOn($decoratedNodeType);
+            }
+            $this->parameters[$decoratedNodeType->getName()] = $decoratedNodeType;
+            $this->parameters[$decoratedNodeType->getSourceEntityFullQualifiedClassName()] = $decoratedNodeType;
+        }
```

Comment:
You should try/catch any Doctrine /PDO exception to avoid errors when DB is not reachable, in tests for example.